### PR TITLE
saga: 7.9.1 -> 8.4.0

### PR DESCRIPTION
**Pull request by wegank**
###### Description of changes

Also fixed build on aarch64-darwin.

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [x] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/gis/saga/default.nix [hunk 1]
```diff
@@ -31,32 +31,29 @@
...
   src = fetchurl {
     url = "mirror://sourceforge/saga-gis/SAGA%20-%20${lib.versions.major version}/SAGA%20-%20${version}/saga-${version}.tar.gz";
-    sha256 = "sha256-Jq1LhBSeJuq9SlNl/ko5I8+jnjZnLMfGYNNUnzVWo7w=";
+    sha256 = "sha256-v6DPwV20fcsznrEaFJk0/ewU4z3cTjzYYuLkyMwSLV0=";
   };
 
+  sourceRoot = "saga-${version}/saga-gis";
+
   nativeBuildInputs = [
     # Upstream's gnerated ./configure is not reliable
```
Comment (ajs124) [inline pkgs/applications/gis/saga/default.nix:44]:
```suggestion
```
This comment doesn't make sense if we're building with cmake.

File: pkgs/applications/gis/saga/default.nix [hunk 2]
```diff
@@ -77,9 +74,11 @@ mkDerivation rec {
     sqlite
   ];
 
-  enableParallelBuilding = true;
+  cmakeFlags = [
+    "-DOpenMP_SUPPORT=${if stdenv.isDarwin then "OFF" else "ON"}"
+  ];
 
-  CXXFLAGS = lib.optionalString stdenv.cc.isClang "-std=c++11 -Wno-narrowing";
+  enableParallelBuilding = true;
```
Comment (ajs124) [inline pkgs/applications/gis/saga/default.nix:81]: isn't this the default with cmake? I might be wrong about this.

**Comment by ajs124** [comment]
this needs a rebase after 30d66019b450e7027a844957d2f0bb1df078950c

File: pkgs/applications/gis/saga/default.nix (hunk 2 above)
Comment (wegank) [reply pkgs/applications/gis/saga/default.nix]: Absolutely. Thanks!

File: pkgs/applications/gis/saga/default.nix (hunk 1 above)
Comment (wegank) [reply pkgs/applications/gis/saga/default.nix]: Removed.

**Pull request merged**